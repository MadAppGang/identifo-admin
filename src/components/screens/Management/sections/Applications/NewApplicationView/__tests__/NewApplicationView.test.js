import { NewApplicationView } from '..';

describe('<NewApplicationView />', () => {
  let props;

  beforeEach(() => {
    props = {
      saving: false,
      error: null,
      postApplication: jest.fn(),
      resetError: jest.fn(),
      history: {
        push: jest.fn(),
      }
    };
  });

  test('renders as expected', () => {
    expect(shallow(<NewApplicationView {...props} />)).toMatchSnapshot();
  });

  test('passes error to ApplicationForm', () => {
    props.error = new Error('error');
    const component = shallow(<NewApplicationView {...props} />);

    expect(component.find('ApplicationForm').prop('error')).toEqual(props.error);
  });

  test('passes saving to ApplicationForm', () => {
    props.saving = true;
    const component = shallow(<NewApplicationView {...props} />);

    expect(component.find('ApplicationForm').prop('loading')).toEqual(props.saving);
  });

  test('calls postApplication on ApplicationForm submit', () => {
    const application = { id: '11' };
    shallow(<NewApplicationView {...props} />)
      .find('ApplicationForm')
      .simulate('submit', application);
    expect(props.postApplication).toBeCalledWith(application);
  });

  test('calls resetError on ApplicationForm cancel', () => {
    shallow(<NewApplicationView {...props} />)
      .find('ApplicationForm')
      .simulate('cancel');
    expect(props.resetError).toBeCalled();
  });

  test('pushes to history on ApplicationForm cancel', () => {
    shallow(<NewApplicationView {...props} />)
      .find('ApplicationForm')
      .simulate('cancel');
    expect(props.history.push).toBeCalledWith('/management/applications');
  });
});
