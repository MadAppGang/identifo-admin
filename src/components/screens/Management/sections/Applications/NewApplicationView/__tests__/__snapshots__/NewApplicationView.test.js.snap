// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewApplicationView /> renders as expected 1`] = `
<section
  className="iap-management-section"
>
  <header>
    <div>
      <Link
        className="iap-management-section__back"
        replace={false}
        to="/management/applications"
      >
        ←  Applications
      </Link>
    </div>
    <p
      className="iap-management-section__title"
    >
      Create Application
    </p>
    <p
      className="iap-management-section__description"
    >
      Configure the allowed Callback URLs and Secrets for your Application.
    </p>
  </header>
  <main>
    <ApplicationForm
      application={null}
      error={null}
      loading={false}
      onCancel={[Function]}
      onSubmit={[MockFunction]}
    />
  </main>
</section>
`;
